### ISSUE TITLE
URL for help in Rose GUI is broken

### ISSUE THREAD

**Issue by jonnyhtw** [metomi/rose#2410]
Hi there

The URL for the Rose help pages from the GUI is broken. I've attached screenshots demonstrating this.

If someone can supply me with the correct URL I can have a go at fixing this if need be.

Thanks 😄 

Jonny

![Screenshot 2020-09-23 at 17 19 36](https://user-images.githubusercontent.com/12940715/93969913-1a4d2400-fdc1-11ea-862b-b8856d51968f.png)
![image](https://user-images.githubusercontent.com/12940715/93969907-16210680-fdc1-11ea-825c-6b3178f32a5e.png)

**Comment by wxtim** [comment]
This isn't going to look great, but I'm not sure that there _is_ a suitable page describing how `rose config-edit` works in ![the most up-to-date rose docs](https://metomi.github.io/rose/2019.01.2/html/search.html?q=rose+config-edit&check_keywords=yes&area=default).

You could put a link to https://metomi.github.io/rose/2019.01.2/html/api/configuration/metadata as the next best thing. Looking at the codebase you'd want to look at `branch=2019.02.1::lib/python/config_editor/__init__.py`. This GUI (Which is based on Python2 libraries) will be re-built at some point, and is only available in Rose 2019.x versions, not in Rose 2.0.

**Comment by jonnyhtw** [comment]
Thanks Tim.

I'll have a go at creating a branch/PR on this...

All the best

Jonny

**Comment by jonnyhtw** [comment]
Hi again

I think the first suggestion that you put here @wxtim would be more suitable; https://metomi.github.io/rose/2019.01.2/html/api/configuration/metadata.

At the moment, the broken link is pointing here, https://metomi.github.io/rose/doc/rose-rug-config-edit.html and is only referred to in the `__int__.py` file here...

![image](https://user-images.githubusercontent.com/12940715/94143185-97f95880-fec3-11ea-817a-3ed1be5026f3.png)

Does this mean that the file holding the info that we want to point to will need to go into the `doc` directory?

Cheers

Jonny

**Comment by wxtim** [comment]
Hi @jonnyhtw 

**Rose Config Editor is written in a Python2 only version of GTK: It is no longer part of the master branch of Rose, and is due to be replaced with a completely new config editor.**
Therefore I wouldn't advocate putting too much time into fixing the docs, which will need a total re-write anyway.

Fixing the link wouldn't be a bad idea: You could just put in a generic link to the Rose doc, or one of the specific pages listed above. I'm happy to have a little look with you at how it constructs the link, but I think your best bet is changing the link. 

I don't know what other devs would think, but at this stage in the lifecycle of config-edit I'd be happy to accept a change that looks like:

```diff
--- a/lib/python/rose/config_editor/window.py
+++ b/lib/python/rose/config_editor/window.py
@@ -261,8 +261,7 @@ class MainWindow(object):
 
     def launch_help_dialog(self, somewidget=None):
         """Launch a browser to open the help url."""
-        webbrowser.open(rose.resource.ResourceLocator.default().get_doc_url() +
-                        rose.config_editor.HELP_FILE, new=True, autoraise=True)
+        webbrowser.open('https://www.google.co.uk', new=True, autoraise=True)
         return False
 
     def launch_ignore_dialog(self, name_section_dict, prefs, is_ignored):
```

**but if you are still interested in writing your own docs**

> Does this mean that the file holding the info that we want to point to will need to go into the doc directory?

No. If you were to write documentation this would go into a subfolder of the `sphinx` dir and you'd use `rose make-docs clean html`. You'd need to have an environment that satisfied `rose check-software --docs`.

**Comment by dpmatthews** [comment]
I think the simplest thing would be change the link to point to the main documentation page:
https://metomi.github.io/rose/doc/html/index.html
Possibly also change the menu item to just say "Help" rather than "GUI Help".

**Comment by jonnyhtw** [comment]
Hi there

I've made a fork of the Rose `2019.01.x` branch [here](https://github.com/jonnyhtw/rose/tree/2019.01.x).

I've also made a pull request which is here ---> #2411 

Just shout in the likely even that I'm doing this wrong; this isn't something I do much! 🤣 

All the best

Jonny

**Comment by jonnyhtw** [comment]
> I think the simplest thing would be change the link to point to the main documentation page:
> 
> https://metomi.github.io/rose/doc/html/index.html
> 
> Possibly also change the menu item to just say "Help" rather than "GUI Help".

Thanks a lot for this!

**Comment by jonnyhtw** [comment]
> Hi @jonnyhtw 
> 
> 
> 
> **Rose Config Editor is written in a Python2 only version of GTK: It is no longer part of the master branch of Rose, and is due to be replaced with a completely new config editor.**
> 
> Therefore I wouldn't advocate putting too much time into fixing the docs, which will need a total re-write anyway.
> 
> 
> 
> Fixing the link wouldn't be a bad idea: You could just put in a generic link to the Rose doc, or one of the specific pages listed above. I'm happy to have a little look with you at how it constructs the link, but I think your best bet is changing the link. 
> 
> 
> 
> I don't know what other devs would think, but at this stage in the lifecycle of config-edit I'd be happy to accept a change that looks like:
> 
> 
> 
> ```diff
> 
> --- a/lib/python/rose/config_editor/window.py
> 
> +++ b/lib/python/rose/config_editor/window.py
> 
> @@ -261,8 +261,7 @@ class MainWindow(object):
> 
>  
> 
>      def launch_help_dialog(self, somewidget=None):
> 
>          """Launch a browser to open the help url."""
> 
> -        webbrowser.open(rose.resource.ResourceLocator.default().get_doc_url() +
> 
> -                        rose.config_editor.HELP_FILE, new=True, autoraise=True)
> 
> +        webbrowser.open('https://www.google.co.uk', new=True, autoraise=True)
> 
>          return False
> 
>  
> 
>      def launch_ignore_dialog(self, name_section_dict, prefs, is_ignored):
> 
> ```
> 
> 
> 
> **but if you are still interested in writing your own docs**
> 
> 
> 
> > Does this mean that the file holding the info that we want to point to will need to go into the doc directory?
> 
> 
> 
> No. If you were to write documentation this would go into a subfolder of the `sphinx` dir and you'd use `rose make-docs clean html`. You'd need to have an environment that satisfied `rose check-software --docs`.

OK got it! I've had a go!

**Comment by jonnyhtw** [comment]
fixed by #2411  🥳
